llvm/lib/IR/Verifier.cpp

```diff
@@ -3604,8 +3604,10 @@ void Verifier::verifyMustTailCall(CallInst &CI) {
 
   // - The caller and callee prototypes must match.  Pointer types of
   //   parameters or return types may differ in pointee type, but not
-  //   address space.
-  if (!CI.getCalledFunction() || !CI.getCalledFunction()->isIntrinsic()) {
+  //   address space. Note that this is not applicable for OCaml functions,
+  //   since their arguments will not be passed from the stack.
+  if ((!CI.getCalledFunction() || !CI.getCalledFunction()->isIntrinsic()) &&
```

Comment:
According to this [doc](https://llvm.org/docs/LangRef.html#calling-conventions), this check is only needed "if the calling convention is not swifttailcc or tailcc". It'd be better to move the new check for `CallingConv::OCaml` to the same place.